src/rgw/rgw_realm_reloader.cc

```diff
@@ -186,9 +186,12 @@ void RGWRealmReloader::reload()
    * the dynamic reconfiguration. */
   env.auth_registry = rgw::auth::StrategyRegistry::create(
       cct, implicit_tenants, env.driver);
-  env.lua.manager = env.driver->get_lua_manager(env.lua.manager->luarocks_path());
-  if (env.lua.background) {
-    env.lua.background->set_manager(env.lua.manager.get());
+  if (!cct->_conf->rgw_lua_enable) {
```

Comment:
there could be an issue here. if you disable and then reload, the lua manager could crash if noit reloaded.
i think it is better to check: `env.lua.manager == nullptr`

Reply:
> the issue is if we start with lua enabled, then we disable it in the conf option (without a restart, so, the lua background is still running), and then we do a realm reload. if we skip these lines the lua background is still holding the handle to the old rados, and wil lcrash when it tries to use it. you should check if the lua manager is not null, if it is not null, you should run this code

but then if i check for `if manager != null`, in your use case `manager is NOT NULL` and hence even with realm reload and lua disabled the code in the loop will execute and run in the background.

so ideally we need to confirm what needs to be done on realm reload ?
isn't realm reload almost same as rgw restart ? if yes then we want to stop the lua background if running and lua is disabled ?

Reply:
AFAIK, the "init_lua()" function is called on when the RGW start. So, a relam reload does not run it, and therefore has no impact on enabling or disabling lua.

Reply:
thats how i understood as well, so then why are was the initialization of lua background thread done again on realm reload ?
if it was running during the start of rgw, it will be running after the relam reload too and vice versa ??

Reply:
we dont restart the thread. just set the new lua manager (which has the new rados handle)

Reply:
@yuvalif so coming back to original comment, the current check of `rgw_lua_enable` should be fine i feel, coz on realm reload the configs are not read as it using the loaded config [here](https://github.com/ceph/ceph/blob/main/src/rgw/rgw_realm_reloader.cc#L89)
so with current check, even if the lua config is enabled before realm reload and rgw is not restarted post changing the value, lua config flag value will still be false and hence the lua_manager will not be re-initialized with new backend store

Reply:
the value of a conf option is updated immediatly (no reload or restart are needed) when you set via the ceph conf tool. e.g. (for vstart):
```
ceph config set client.rgw.8000 <option name> <option value>
```

Reply:
done.